language/move-vm/runtime/src/runtime.rs

```diff
@@ -374,7 +374,12 @@ impl VMRuntime {
         let mut dummy_locals = Locals::new(params.len());
         // index and (inner) type of mutable ref inputs. we will use them to return the effects of `func` on these inputs
         let mut mut_ref_inputs = Vec::new();
-        for (idx, (arg, arg_type)) in args.into_iter().zip(params).enumerate() {
+        for (idx, (arg, mut arg_type)) in args.into_iter().zip(params).enumerate() {
+            if let Type::TyParam(_) = arg_type {
```

Comment:
This seems reasonable. Is there a reason we didn't do this for normal (script) entry points? Just fear or something else?

Reply:
Good question..

In normal script entrypoints, we don't allow (e.g.) `foo<T>(t: T)` as an entrypoint, someone can instantiate T with (e.g.) `Coin<USD>` and bytes that happen to serialize to an object of that type. This is effectively minting out of thin air (more generally, this violates the integrity of type constructors).

So adapters that use this API should be cautious about this sort of thing. But of course, this API also lets entrypoints take a `Coin<USD>` directly, and adapters also have to be careful about that.

Reply:
Sure sure. Maybe we could allow this for scripts as long as the type argument followed the  rest of the normal rules? (basically no structs, at least under the current rules)

Reply:
Yep, I think that would work! I have often wanted a `prim` (or similar) ability that every non-struct value has--if that existed, we could very concisely describe your scheme as "Allow generic parameters of type `T` where `T: prim`".

Reply:
Yeah there are a bunch of constraints that could be interesting like that (`private` comes to mind too). Just gets soupy and possibly confusing after a while...